dash/include/dash/atomic/Operation.h

```diff
@@ -31,15 +31,15 @@ namespace atomic {
  * Get the value of the shared atomic.
  */
 template<typename T>
-T load(const dash::GlobRef<dash::Atomic<T>> & ref){
+T load(const dash::GlobRef<dash::Atomic<T>>& ref){
   return ref.load();
 }
 
 /**
- * Set the value of the atomic reference 
+ * Set the value of the atomic reference
  */
 template<typename T>
-void store(const dash::GlobRef<dash::Atomic<T>> & ref,
+void store(dash::GlobRef<dash::Atomic<T>> ref,
```

Comment:
Cant we use nonconst references here? Or do we better rely on compiler optimization?

Reply:
Unfortunately not, because we then cannot pass an rvalue in, e.g.,

```
dash::Array<dash::Atomic<int>> arr(100);
dash::store(arr[1], 1);
```

Reply:
Sorry for asking again, but why can't we use universal references? This would work:
```cpp
void store(dash::GlobRef<dash::Atomic<T>> && ref,...);
```

Reply:
Ahh, totally forgot about that. Fix it :+1:

Reply:
No!
As soon as a type is partially restricted, it is not a forwarding reference any longer. It's a plain overload for a temporary (aka rvalue).

~~~c++
template <typename T> void func(T && param); // <-- forwarding ref
template <typename T> void func(Foo<T> && param); // <-- rvalue
template <typename T> void func(const T && param); // <-- whisky tango foxtrott? ... but still an rvalue ref
~~~

Also, it's "forwarding reference". Meyers used the term "universal reference" in a great article to formulate the type deduction mechanism, but the terminus technicus has be standardized as "forwarding reference".

See:
- https://isocpp.org/blog/2012/11/universal-references-in-c11-scott-meyers
- https://blog.petrzemek.net/2016/09/17/universal-vs-forwarding-references-in-cpp/

Reply:
To make this a helpful comment: what do you suggest?